### DISCUSSION TITLE
dev-ветка

### DISCUSSION THREAD

**Discussion by solarrust** [doka-guide/platform#284] [Идеи]
Есть мысль вести разработку в dev-ветке, а затем вмёрдживать релиз в мейн, чтобы не внедрять постоянно мелкие улучшения на платформе, тем самым по несколько раз на дню меняя внешний вид доки и пугая пользователей. Плюс дополнительная возможность потестить новую фичу в полностью рабочем окружении, без страха выкатить баг на рабочую версию.

**Comment by igsekor** [comment]
Кажется, что кроме неудобства пользователя, который может даже в процессе чтения статьи, открытия перекрёстных ссылок и обновления страницы столкнуться с новым интерфейсом, важно не тормозить процесс разработки. Можно деплоить текущую версию на `dev.doka.guide` или на Surge...

**Comment by pepelsbey** [comment]
По сути, это подразумевает создание трёх шагов:

1. Ветка с пул-реквестом
2. Дев-ветка после мёржа
3. Мейн-ветка после после мёржа

Кажется длинновато и не очень понятно кто, как и когда переносит всё из дева и мейн. Я бы предложил оставить два шага:

1. Ветка с пул-реквестом, где поднимается одноразовый тестовый сервер на Surge
2. Мейн-ветка после мёржа.

И тогда мёрж в пул-реквесте станет тем самым триггером для попадания на сайт. А уж как часто мы будем мёржить эти пулреквесты — это уже можем обсудить отдельно.

Мы можем позволить себе не иметь стейджинга или дева — у нас статика и мы разворачиваемся на любом временном бэкенде простым копированием файлов.

**Reply by monochromer** [reply]
Это хорошо, когда вкатываются мелкие независимые функции. Но если вкатывать все PR по платформе сейчас как есть, то сайт будет выглядеть немного странно.

**Reply by pepelsbey** [reply]
Думаю, мы сейчас больше о том, как мы будем двигаться после запуска. Сейчас мы можем как угодно мёржить — мы никому не мешаем.
